colossalai/utils/profiler/comm_profiler.py

```diff
@@ -101,6 +111,30 @@ def reset(self):
         self.pending_metadata = None
         self.warn_flag = False
 
+    def enable(self):
+        global COL_COMM_PROF
+        COL_COMM_PROF = self

```

Comment:
seems strange to me, is this a better way to write this line?

```python
def all_reduce(profiler: CommProfier,
                    tensor: torch.Tensor,
                    op: ReduceOp = ReduceOp.SUM,
                    group=None,
                    async_op: bool = False,
                         ):
        ...


dist.all_reduce = functools.partial(all_reduce, profiler=self)

```

Reply:
Done